templates/slim/block_image.html.slim

```diff
@@ -1,10 +1,21 @@
-.imageblock(id=@id class=[@style,role]
-    style=[("text-align: #{attr :align}" if attr? :align),("float: #{attr :float}" if attr? :float)].compact.join('; '))
-  .content
+- width = (attr? :width) ? (attr :width) : nil
+- height = (attr? :height) ? (attr :height) : nil
+
+/ When the stretch class is present, block images will take the most space
+/ they can take. Setting width and height can override that.
+/ We pinned the 100% to height to avoid aspect ratio breakage and since
+/ widescreen monitors are the most popular, chances are that height will
+/ be the biggest constraint
+- if (has_role? 'stretch') && !((attr? :width) || (attr? :height))
+  - height = "100%"
+
+- unless attr(1) == 'background' || attr(1) == "canvas"
+  .imageblock(id=@id class=[@style,role]
```

Comment:
I'd like to see modern HTML used for images like what's in the Bespoke converter (i.e., figure/figcaption). See https://github.com/asciidoctor/asciidoctor-bespoke/blob/master/templates/slim/image.html.slim In general, we should be gravitating towards modern HTML which also works best with Reveal.js. We have to strike a balance.